publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -375,11 +378,28 @@ public void updateResource(Resource resource) {
     // update this method according to current needs.
     public Entity migrate(Entity dataEntry) {
         if (isResourceWithFiles(dataEntry)) {
+            persistLogEntriesIfNeeded((Resource) dataEntry);
             return migrateResourceWithFiles((Resource) dataEntry);
         }
         return dataEntry;
     }
 
+    private void persistLogEntriesIfNeeded(Resource resource) {
+        var userInstance = UserInstance.fromPublication(resource.toPublication());
+        var logEntries = resource.fetchLogEntries(this);
+        if ("nve@5948.0.0.0".equals(resource.getResourceOwner().getUser().toString())) {
+            resource.setResourceEvent(ImportedResourceEvent.fromImportSource(ImportSource.fromBrageArchive("NVE"),
+                                                                   resource.getCreatedDate()));
+        } else if (PUBLISHED.equals(resource.getStatus()) && logEntries.stream().noneMatch(entry -> LogTopic.PUBLICATION_PUBLISHED.equals(entry.topic()))) {
+            var publishedDate = Optional.of(resource)
+                                    .map(Resource::getPublishedDate)
+                                    .orElse(resource.getCreatedDate());
+            resource.setResourceEvent(PublishedResourceEvent.create(userInstance, publishedDate));
+        } else if (!PUBLISHED.equals(resource.getStatus()) && logEntries.stream().noneMatch(entry -> LogTopic.PUBLICATION_CREATED.equals(entry.topic()))) {
+            resource.setResourceEvent(CreatedResourceEvent.create(userInstance, resource.getCreatedDate()));
```

Comment:
You've probably already told me how, but to be sure, this fixes todays exsisting stuff (by adding published/created  log entries) and the imported brage nve stuff (by adding importet log entry). How to we handle future imported brage stuff? I would guess future brage stuff would need the same functionality?

Reply:
Log entries for future imports will be persisted by consuming database events.